### PR TITLE
ref(server): Move all heavy actors into dedicated arbiters

### PR DESCRIPTION
**Pull request by jan-auer**
Each actix arbiter spawns a thread with its own single-threaded event loop. This
allows to create isolation between actors that may cause CPU spikes when
backlogged, and makes fair scheduling in actix easier.

While some of our actors, notably the `OutcomeProducer` and (metrics)
`Aggregator` had been spawned into a dedicated arbiter, most of the other actors
were still running on the main arbiter. The `EnvelopeManager` is known to cause
an increase in CPU consumption due to how it spawns futures internally. This
slows down the project cache and healthcheck, which in turn blocks incoming HTTP
requests.

Since there's a global slowdown of the main arbiter, it becomes hard to separate
cause from effect when investigating slow response times. It is not clear
whether the project cache causes increased queue sizes or whether increased
queue sizes delay project fetching.

To create clear separation, this PR introduces a dedicated arbiter for every
heavy actor.

### CODE REVIEW

File: relay-server/src/service.rs [hunk 1]
```diff
@@ -135,6 +135,9 @@ impl ServiceState {
         let outcome_producer = Arbiter::start(|_| outcome_producer);
         registry.set(outcome_producer.clone());
 
+        let outcome_aggregator = OutcomeAggregator::new(&config, outcome_producer.recipient());
+        registry.set(outcome_aggregator.start());
```
Comment (jjbayer) [inline relay-server/src/service.rs:139]: Two questions:
1. Out of curiosity, why did you move this up here?
2. Why does the outcome aggregator not get its own thread?

**Review by jjbayer — APPROVED** [review]

**Review by flub — APPROVED** [review]

File: relay-server/src/service.rs (hunk 1 above)
Comment (jan-auer) [reply relay-server/src/service.rs:139]: That probably wouldn't have been necessary, and I'm happy to revert that. I merely moved it up so that it's next to the outcome producer.

It doesn't have its own thread since it looks like it does relatively cheap stuff internally. I'm not yet convinced that we should have one thread for each actor/service, so I'm mostly going by what we know causes issues.
